django/template/loaders/cached.py

```diff
@@ -26,12 +27,23 @@ def __init__(self, engine, loaders):
     def get_contents(self, origin):
         return origin.loader.get_contents(origin)
 
+    def _raise_if_exception(self, value, template_name):
+        """
+        Take a value pulled from the template cache, and raise an appropriate
+        exception. If template debugging is off, the cached item will be the
+        raw TemplateDoesNotExist class, otherwise it will be an instance of
+        TemplateDoesNotExist that we need to copy before raising.
+        """
+        if isinstance(value, type) and issubclass(value, TemplateDoesNotExist):
+            raise value(template_name)
+        elif isinstance(value, TemplateDoesNotExist):
+            raise copy_exception(value)
+
     def get_template(self, template_name, template_dirs=None, skip=None):
         key = self.cache_key(template_name, template_dirs, skip)
         cached = self.get_template_cache.get(key)
         if cached:
-            if isinstance(cached, TemplateDoesNotExist):
-                raise cached
+            self._raise_if_exception(cached, template_name)
```

Comment:
Not sure that a separate method here has much advantage. What was your thinking about that?